ros/processor/inventory_events_consumer.py

```diff
@@ -119,7 +119,7 @@ def process_system_details(self, msg):
         """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
         host = msg['host']
         with app.app_context():
-            if msg.get('type') == 'updated':
+            if msg.get("platform_metadata") is None:  # Handles valid update for a ROS system
```

Comment:
```suggestion
            if msg.get('type') == 'updated' AND msg.get("platform_metadata") is None:  # Handles valid update for a ROS system
```

Reply:
Both conditions are being checked [here](https://github.com/RedHatInsights/ros-backend/pull/324/files#diff-73b796983793ca68d788f97a84b00ef19df5157d5fee7ee854c47ed1133d45f1R274-R275)
I had to add `if msg.get("platform_metadata") is None` again because we are handling create and update in one function

Reply:
I still prefer to add this type check here as well.

Reply:
Come to think about it, suggested change is more readable, change pushed